akka-stream/src/main/scala/akka/stream/impl/FanIn.scala

```diff
@@ -186,7 +186,7 @@ private[akka] object FanIn {
     }
 
     def inputsAvailableFor(id: Int) = new TransferState {
-      override def isCompleted: Boolean = depleted(id)
+      override def isCompleted: Boolean = depleted(id) || cancelled(id)
```

Comment:
No, because there is value in distinguishing whether we cancelled it or the outside completed (think inputsOrCompleteAvailableFor)